app/api/routes/resource_modification.py

```diff
@@ -49,8 +53,19 @@ def update_resource(id, json, db):
             resource.languages = langs
             index_object['languages'] = resource.serialize['languages']
         if json.get('category'):
+            old_category = resource.category #I'm assuming this is the category name as a string
```

Comment:
This is not the category name as a string. This is a `Category` object. This is the definition of a `Category` object:
https://github.com/OperationCode/resources_api/blob/a78426a89e4625a733763c4dc24c978d639f8c52/app/models.py#L106-L130 

To get the string representation of the category, you'll want `resource.category.name`